src/us/kbase/workspace/database/Workspace.java

```diff
@@ -836,20 +837,15 @@ private void sortObjects(
 			reports.put(wo, rep);
 			idhandler.associateObject(new IDAssociation(objcount, true));
 			try {
-				for (final Provenance.ProvenanceAction action: wo.getProvenance().getActions()) {
+				for (final ProvenanceAction action: wo.getProvenance().getActions()) {
 					for (final String pref: action.getWorkspaceObjects()) {
-						if (pref == null) {
```

Comment:
provenance refs are now parsed earlier in the stack, so this error is impossible